client/landing/stepper/declarative-flow/internals/steps-repository/woo-install-plugins/index.tsx

```diff
@@ -1,25 +1,106 @@
 /* eslint-disable no-console */
-import { useDispatch } from '@wordpress/data';
+import config from '@automattic/calypso-config';
+import { useDispatch, useSelect } from '@wordpress/data';
 import { useEffect } from 'react';
-import { ONBOARD_STORE } from '../../../../stores';
+import { useSite } from 'calypso/landing/stepper/hooks/use-site';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { logToLogstash } from 'calypso/lib/logstash';
+import { ONBOARD_STORE, SITE_STORE } from '../../../../stores';
 import type { Step } from '../../types';
+import type { AtomicSoftwareStatus } from 'calypso/../packages/data-stores/src/site';
 
 const wait = ( ms: number ) => new Promise( ( res ) => setTimeout( res, ms ) );
 
+export interface FailureInfo {
+	type: string;
+	code: number | string;
+	error: string;
+}
+
 const WooInstallPlugins: Step = function WooInstallPlugins( { navigation } ) {
 	const { submit } = navigation;
-	const { setPendingAction, setProgressTitle } = useDispatch( ONBOARD_STORE );
+	const { setPendingAction, setProgressTitle, setProgress } = useDispatch( ONBOARD_STORE );
+	const { initiateSoftwareInstall, requestAtomicSoftwareStatus } = useDispatch( SITE_STORE );
+	const { getAtomicSoftwareInstallError, getAtomicSoftwareStatus, getAtomicSoftwareError } =
+		useSelect( ( select ) => select( SITE_STORE ) );
+	const site = useSite();
+	const softwareSet = 'woo-on-plans';
+
+	const handleTransferFailure = ( failureInfo: FailureInfo ) => {
+		recordTracksEvent( 'calypso_woocommerce_dashboard_snag_error', {
+			action: failureInfo.type,
+			site: site?.URL,
+			code: failureInfo.code,
+			error: failureInfo.error,
+		} );
+
+		logToLogstash( {
+			feature: 'calypso_client',
+			message: failureInfo.error,
+			severity: config( 'env_id' ) === 'production' ? 'error' : 'debug',
+			blog_id: site?.ID,
+			properties: {
+				env: config( 'env_id' ),
+				type: 'calypso_woocommerce_dashboard_snag_error',
+				action: failureInfo.type,
+				site: site?.URL,
+				code: failureInfo.code,
+			},
+		} );
+	};
 
 	useEffect( () => {
+		if ( ! site?.ID ) return;
+
 		setPendingAction( async () => {
-			setProgressTitle( 'Installing plugins' );
-			await wait( 3000 );
-			// TODO actually install plugins
+			setProgressTitle( 'Installing WooCommerce' );
```

Comment:
Together with setting the title, we should enable the progress bar (instead of the ellipsis) by  using 
`setProgress( 0 );`
I didn't have it on the transfer step as well and noticed the ellipsis being there before the bar :)